api/v1/verticaautoscaler_webhook.go

```diff
@@ -339,6 +340,34 @@ func (v *VerticaAutoscaler) validateScaledObject(allErrs field.ErrorList) field.
 	return allErrs
 }
 
+func (v *VerticaAutoscaler) validatePrometheusAuthModes(allErrs field.ErrorList) field.ErrorList {
+	authModeTypes := []PrometheusAuthModes{PrometheusAuthBasic, PrometheusAuthBearer, PrometheusAuthCustom, PrometheusAuthTLS, PrometheusAuthTLSAndBasic}
+	pathPrefix := field.NewPath("spec").Child("customAutoscaler")
+	if v.IsScaledObjectEnabled() {
+		for i := range v.Spec.CustomAutoscaler.ScaledObject.Metrics {
+			metric := &v.Spec.CustomAutoscaler.ScaledObject.Metrics[i]
+			// validate if authSecert is set, AuthModes can not be empty
+			if metric.Prometheus != nil && metric.AuthSecret != "" && metric.Prometheus.AuthModes == "" {
+				err := field.Invalid(pathPrefix.Child("scaledObject").Child("metrics").Index(i).Child("prometheus").Child("authModes"),
+					v.Spec.CustomAutoscaler.ScaledObject.Metrics[i].Prometheus.AuthModes,
+					"When authSecret is set authModes can not be empty.",
+				)
+				allErrs = append(allErrs, err)
+			}
+			// validate AuthModes type if set
+			if metric.Prometheus != nil && metric.Prometheus.AuthModes != "" && !slices.Contains(authModeTypes, metric.Prometheus.AuthModes) {
```

Comment:
```suggestion
			if metric.Prometheus != nil && metric.AuthSecret != "" metric.Prometheus.AuthModes != "" && !slices.Contains(authModeTypes, metric.Prometheus.AuthModes) {
```
If authSecret is empty, we do not need this check as authModes will be ignored.

Reply:
Good call, addressed in https://github.com/vertica/vertica-kubernetes/pull/1098/commits/165a59d235e2f76815d87eca0255d1ff1e527200